scrapers/src/scrapers/co/sos/general_candidates.rs

```diff
@@ -0,0 +1,134 @@
+use std::error::Error;
+
+use scraper::{Html, Selector};
+
+use crate::{extractors::*, util, util::extensions::NoneIfEmptyExt};
+
+const HTML_PATH: &'static str = "co/sos/general_candidates.html";
+const PAGE_URL: &'static str =
+    "https://www.sos.state.co.us/pubs/elections/vote/generalCandidates.html";
+
+#[derive(Default)]
+pub struct Scraper {}
+
+impl crate::Scraper for Scraper {
+    async fn run(&self, context: &crate::ScraperContext<'_>) -> Result<(), Box<dyn Error>> {
+        let html = reqwest::get(PAGE_URL).await?.text().await?;
+        Self::scrape_html(html, context).await
+    }
+
+    async fn run_local(&self, context: &crate::ScraperContext<'_>) -> Result<(), Box<dyn Error>> {
+        let html = util::read_local_html(&HTML_PATH)?;
+        Self::scrape_html(html, context).await
+    }
+}
+
+impl Scraper {
+    pub async fn scrape_html(
+        html: String,
+        context: &crate::ScraperContext<'_>,
+    ) -> Result<(), Box<dyn Error>> {
+        let entries = Self::parse_raw_entries(html)?;
+        for entry in entries {
+            let mut office = Self::build_office_input(&entry);
+            office.slug = Some(crate::generate_office_slug(&office));
+            if let Err(err) = db::Office::upsert_from_source(&context.db.connection, &office).await
+            {
+                // TODO - Track/log error
```

Comment:
Thanks for flagging this, any ideas on best approach for visibility/observability? Maybe I'll spin up a slackbot or something

Reply:
I think a slack bot would be a good place to start. But we may want to do something a little more sophisticated than just fire a message for every `Err` we get. I'm kinda thinking we'll want to aggregate the failures we receive during a job, then send a message like "Job XYZ failed to capture 5 out of 143 candidate records". Open to other ideas as well.

Reply:
I think the aggregate approach makes sense, any log of what went wrong will be super useful for fixing and fine tuning these, esp if pages change structure